### PR TITLE
image: introduce an ImageKind abstraction

### PR DESCRIPTION
**Pull request by teg**
An ImageKind encapsulates how to generate a manifest for a certain kind of image. Each image type in the distro packages
will be implemented in terms of an image kind, by instantiating its configuration and calling the `InstantiateManifest`
function, which does the actual pipeline creation and instantiation. Different image kinds will have different configuration and
produce different pipelines.

The `osbuild-playground` command is reimplemented in terms of ImageKinds, ideally making it easy to create new ones
and move them over to `internal/image` when satisfied. Note that `osbuild-playground` takes its input as a JSON serialised
struct, but probably anything in `internal/image` would use more high-level objects than that, so might not be a straight
copy-paste.

Furthermore, the fedora image definitions have been ported over to be generated in this way. There is a lot of copying of
struct fields going on in distro/fedora/manifests.go now, and ideally we would not have that indirection but instantiate things directly in the image definitions, at least to a greater extent, so there is a lot of room for future work.

An unsolved problem is how to properly expose the artefact objects from the `InstantiateManifest` function, we probably
want to return a struct of these per image kind, but that might require some rework.

### CODE REVIEW

File: internal/manifest/raw.go [hunk 1]
```diff
@@ -66,3 +67,8 @@ func (p *RawImage) serialize() osbuild2.Pipeline {
 
 	return pipeline
 }
+
+func (p *RawImage) Export() *artefact.Artefact {
+	p.Base.export = true
```
[the remark this answers is not in the document]
Comment (teg) [reply internal/manifest/raw.go:72]: Yup!

File: internal/manifest/os.go [hunk 2]
```diff
@@ -92,6 +87,22 @@ type OS struct {
 	SshdConfig    *osbuild2.SshdConfigStageOptions
 	AuthConfig    *osbuild2.AuthconfigStageOptions
 	PwQuality     *osbuild2.PwqualityConfStageOptions
+}
+
+// OS represents the filesystem tree of the target image. This roughly
+// correpsonds to the root filesystem once an instance of the image is running.
```
[the remark this answers is not in the document]
Comment (teg) [reply internal/manifest/os.go:93]: Fixed!

File: internal/artefact/artefact.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package artefact
```
[the remark this answers is not in the document]
Comment (teg) [reply internal/artefact/artefact.go]: Fixed.

**Review by gicmo — DISMISSED** [review]
❤️

**Comment by gicmo** [comment]
This needs a rebase.

**Comment by teg** [comment]
Rebased :)

**Review by achilleas-k — APPROVED** [review]

**Review by gicmo — APPROVED** [review]
❤️

**Pull request merged**